### PR TITLE
porpagate flags into lua filters

### PR DESCRIPTION
**Pull request by KSmigielski**

### CODE REVIEW

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactory.kt [hunk 1]
```diff
@@ -96,6 +96,18 @@ class LuaFilterFactory(private val incomingPermissionsProperties: IncomingPermis
                         Value.newBuilder()
                             .setStringValue(group.discoveryServiceName ?: "")
                             .build()
+                    ).putFields("flags", Value.newBuilder()
+                        .setStructValue(Struct.newBuilder()
```
Comment (malfoj89) [inline envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactory.kt:100]: maybe we should add test for this to ensure it works as desired

**Review by malfoj89 — CHANGES_REQUESTED** [review]

File: envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt [hunk 2]
```diff
@@ -37,4 +38,42 @@ internal class LuaFilterFactoryTest {
...
+    fun `should leave flags empty`() {
+        val expectedServiceName = "service-1"
+        val expectedDiscoveryServiceName = "consul-service-1"
+        val group: Group = ServicesGroup(
+            communicationMode = CommunicationMode.XDS,
+            serviceName = expectedServiceName,
+            discoveryServiceName = expectedDiscoveryServiceName
+        )
+        val factory = LuaFilterFactory(IncomingPermissionsProperties())
+        // when
```
Comment (malfoj89) [inline envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt:74]: please add line break as in above test

File: envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt [hunk 3]
```diff
@@ -37,4 +38,42 @@ internal class LuaFilterFactoryTest {
...
+        val luaMetadata = metadata.getFilterMetadataOrThrow("envoy.filters.http.lua")
+        // then
+        flags.forEach { (k, v) ->
+            assertThat(luaMetadata.getFieldsOrThrow("flags").structValue.getFieldsOrThrow(k).boolValue).isEqualTo(v)
+        }
+    }
+
+    @Test
+    fun `should leave flags empty`() {
+        val expectedServiceName = "service-1"
```
Comment (malfoj89) [inline envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt:66]: missing // given

File: envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt [hunk 4]
```diff
@@ -37,4 +38,42 @@ internal class LuaFilterFactoryTest {
...
+        val metadata = factory.ingressScriptsMetadata(group, flags)
+        val luaMetadata = metadata.getFilterMetadataOrThrow("envoy.filters.http.lua")
+        // then
+        flags.forEach { (k, v) ->
+            assertThat(luaMetadata.getFieldsOrThrow("flags").structValue.getFieldsOrThrow(k).boolValue).isEqualTo(v)
+        }
+    }
+
+    @Test
+    fun `should leave flags empty`() {
```
Comment (malfoj89) [inline envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt:65]:
```suggestion
    fun `should setup flags empty if not specified`() {
```

File: envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt [hunk 5]
```diff
@@ -37,4 +38,42 @@ internal class LuaFilterFactoryTest {
...
+        val group: Group = ServicesGroup(
+            communicationMode = CommunicationMode.XDS,
+            serviceName = expectedServiceName,
+            discoveryServiceName = expectedDiscoveryServiceName
+        )
+        val factory = LuaFilterFactory(IncomingPermissionsProperties())
+        // when
+        val metadata = factory.ingressScriptsMetadata(group)
+        val luaMetadata = metadata.getFilterMetadataOrThrow("envoy.filters.http.lua")
+        assertThat(luaMetadata.getFieldsOrThrow("flags").structValue.allFields).isEmpty()
```
Comment (malfoj89) [inline envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt:77]: please add // then above

File: envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt [hunk 6]
```diff
@@ -37,4 +38,42 @@ internal class LuaFilterFactoryTest {
...
+        val expectedServiceName = "service-1"
+        val expectedDiscoveryServiceName = "consul-service-1"
+        val group: Group = ServicesGroup(
+            communicationMode = CommunicationMode.XDS,
+            serviceName = expectedServiceName,
+            discoveryServiceName = expectedDiscoveryServiceName
+        )
+        val factory = LuaFilterFactory(IncomingPermissionsProperties())
+        // when
+        val metadata = factory.ingressScriptsMetadata(group)
```
Comment (malfoj89) [inline envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt:75]: i believe it belongs to given part as when part shall have only desired interaction

File: envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt [hunk 7]
```diff
@@ -37,4 +38,42 @@ internal class LuaFilterFactoryTest {
...
+            communicationMode = CommunicationMode.XDS,
+            serviceName = expectedServiceName,
+            discoveryServiceName = expectedDiscoveryServiceName
+        )
+        val factory = LuaFilterFactory(IncomingPermissionsProperties())
+
+        // when
+        val metadata = factory.ingressScriptsMetadata(group, flags)
+        val luaMetadata = metadata.getFilterMetadataOrThrow("envoy.filters.http.lua")
+        // then
```
Comment (malfoj89) [inline envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt:58]: add line break above // then

File: envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt [hunk 8]
```diff
@@ -37,4 +38,42 @@ internal class LuaFilterFactoryTest {
...
+        val expectedDiscoveryServiceName = "consul-service-1"
+        val group: Group = ServicesGroup(
+            communicationMode = CommunicationMode.XDS,
+            serviceName = expectedServiceName,
+            discoveryServiceName = expectedDiscoveryServiceName
+        )
+        val factory = LuaFilterFactory(IncomingPermissionsProperties())
+
+        // when
+        val metadata = factory.ingressScriptsMetadata(group, flags)
```
Comment (malfoj89) [inline envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt:56]: belong to given

**Review by malfoj89 — CHANGES_REQUESTED** [review]

**Review by malfoj89 — APPROVED** [review]

**Review by gawarz — APPROVED** [review]

**Pull request merged**